src/fidesapi/routes/generate.py

```diff
@@ -0,0 +1,153 @@
+    type: GenerateTypes
+
+
+class GenerateRequestPayload(BaseModel):
+    """
+    The model for the request body housing generate information.
+    """
+
+    organization_key: str
+    generate: Generate
+
+
+class GeneratedResponse(BaseModel):
+    """
+    The model to hous the response for generated infrastructure.
+    """
+
+    generate_results: Optional[List[System]]
+
+
+router = APIRouter(tags=["Generate"], prefix=f"{API_PREFIX}/generate")
+
+
+@router.post(
+    "/",
+    status_code=status.HTTP_200_OK,
+    response_model=GeneratedResponse,
+    responses={
+        status.HTTP_400_BAD_REQUEST: {
+            "content": {
+                "application/json": {
+                    "example": {"detail": "Unable to generate resources."}
+                }
+            }
+        },
+    },
+)
+async def generate(
+    generate_request_payload: GenerateRequestPayload, response: Response
+) -> Dict:
+    """
+    A multi-purpose endpoint to generate Fides resources based on existing
+    infrastructure
+
+    Currently generates Fides resources for the following:
+    * AWS: Systems
+
+    In the future, this will include options for other Systems & Datasets,
+    examples include:
+    * Okta: Systems
+    * Snowflake: Datasets
+
+    All config secrets should be encoded as a minor security precaution, using the
+    `obscure_string` function in `fidesapi.routes.util`
+
+    All production deployments should implement HTTPS for security purposes
+    """
+    organization = await get_resource(
+        sql_model_map["organization"], generate_request_payload.organization_key
+    )
+    if generate_request_payload.generate.target.lower() == "aws":
+        try:
+            import boto3  # pylint: disable=unused-import
+        except ModuleNotFoundError:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Packages not found, ensure aws is included: fidesctl[aws]",
+            )
+        log.info("Setting config for AWS")
+        generated_systems = generate_aws(
+            aws_config=AWSConfig(**generate_request_payload.generate.config.dict()),
+            organization=organization,
+        )
+    return {"generate_results": generated_systems}
+
+
+def generate_aws(
+    aws_config: AWSConfig, organization: Organization
+) -> List[Dict[str, str]]:
+    """
+    Returns a list of Systems found in AWS.
+    """
+    from botocore.exceptions import ClientError
+
+    config = {
+        "region_name": aws_config.region_name,
+        "aws_access_key_id": unobscure_string(aws_config.aws_access_key_id),
+        "aws_secret_access_key": unobscure_string(aws_config.aws_secret_access_key),
+    }
+    try:
+        log.info("Generating systems from AWS")
+        aws_systems = generate_aws_systems(organization=organization, aws_config=config)
+    except ClientError as error:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=error)
```

Comment:
If the AWS client throws a ClientError, it's not necessarily the case that we should return a 400 to the client. Pydantic is already validating that our `config` objects have the required fields so a ClientError possibly means a bug in our code or that AWS is returning 500s. 

It might be a big ask of this PR to do the error handling even just for boto. I think we'd have to do a deep dive into what kind of responses we can get from the client. 

Here's an example from https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html
```
import botocore
import boto3

client = boto3.client('sqs')
queue_url = 'SQS_QUEUE_URL'

try:
    client.send_message(QueueUrl=queue_url, MessageBody=('some_message'))

except botocore.exceptions.ClientError as err:
    if err.response['Error']['Code'] == 'InternalError': # Generic error
        # We grab the message, request ID, and HTTP code to give to customer support
        print('Error Message: {}'.format(err.response['Error']['Message']))
        print('Request ID: {}'.format(err.response['ResponseMetadata']['RequestId']))
        print('Http code: {}'.format(err.response['ResponseMetadata']['HTTPStatusCode']))
    else:
        raise err
```

Our handling of boto ClientError will probably require looking at `err.response['Error']['Code']` or `err.response['ResponseMetadata']['HTTPStatusCode']`. 

Examples using error code:
InternalError -> Maps to 5xx from fidesctl
AuthFailure -> Maps to 403 from fidesctl with custom message that auth failed with aws.
ValidationError -> Ambigious to me since this could be caused by user input. My first thought is that this should return a 5xx because we should do our own validation before passing requests to aws. 

Usage of HTTPStatusCode might be simpler because maybe we can check for 403s and then any other 400s or 500s we return our own 5xx.

Reply:
Totally agree, love what you have laid out above 💙  

I was sort of holding off on the deep dive into the botocore exceptions due to the scale of depth we could go to and backed off trying to be too specific. `ClientError` as a general starting point seemed like a (weak/overarching) place to start with then coming back to explore more, maybe alongside (timewise) some of the front end work.

This sets some really nice groundwork though so am down to get after it in here, thanks for the thoughtfulness around it all!